### ISSUE TITLE
`@V` state element (aka `$` in ABOP) isn't rendering out correctly

### ISSUE THREAD

**Issue by heckj** [heckj/Lindenmayer#27]
Fundamentally, the example trees rendered with the L-systems taken directly from The Algorithmic Beauty of Plants aren't resulting the same visualizations as described in the book. They're close, but there's complication - I think - around a specific symbol and what it's doing: `@V` in the implementation research paper, and `$` in the book itself.

As an example, compare the results for the following sympodial and monopodial trees:
<img width="1019" alt="lindenmayer2" src="https://user-images.githubusercontent.com/43388/148631760-c3c1651c-7967-4e7e-b71d-c764890e729c.png">
<img width="822" alt="original2" src="https://user-images.githubusercontent.com/43388/148631762-74ed7ec3-97b1-4361-a158-801cf8dace1a.png">
<img width="1023" alt="lindenmayer1" src="https://user-images.githubusercontent.com/43388/148631763-32850ab2-ddb1-40a5-be18-7c0d13ddf5bd.png">
<img width="685" alt="original1" src="https://user-images.githubusercontent.com/43388/148631764-48a8e641-65c3-4acf-92be-bd1b36f9de52.png">

I suspect part of the core of this issue is that what I'm doing to create the rotations for the `$` symbol is resulting in a rotation that is far too vertically oriented.

I thought perhaps I had confused basic transforms and headings, but in reviewing the relevant code (https://github.com/heckj/Lindenmayer/blob/main/Sources/Lindenmayer/SceneKitRenderer.swift#L181-L214), I seem to have accounted for the corrected heading.

Since I've been mentally confusing directions, I suspect that I need to add a debugging view (#26), as well as potentially tests, to generate the relevant objects and then interrogate them to verify rotations, translations, etc.

**Comment by heckj** [comment]
It's still not 100% to what the results from the book show, but it's notably better - so committing this as the fix for the `@V` symbol.
<img width="1101" alt="Screen Shot 2022-01-17 at 1 13 04 PM" src="https://user-images.githubusercontent.com/43388/149837777-8caf68c7-47cd-44e5-8761-d51763da3c2c.png">
<img width="1038" alt="Screen Shot 2022-01-17 at 1 12 01 PM" src="https://user-images.githubusercontent.com/43388/149837796-a386076a-a62a-47a1-87c9-404d95fb9d7d.png">

**Comment by heckj** [comment]
well darnit. The fix is notably improved, but needs more work. While iterating on a debugging view of the Sympodial trees, it appears to be over-rotating just slightly, meaning that something in the calculated angle that is applied to "roll vertical" is incorrect. 

State 119 of the basic sympodial structured tree at 4 iterations illustrates this:

The starting point is:
<img width="359" alt="Screen Shot 2022-01-18 at 2 31 15 PM" src="https://user-images.githubusercontent.com/43388/150029192-416a55af-8ed3-47b3-8074-c010a066ed04.png">

And after application of the "rollUpToVertical" (position 120), the state is:
<img width="357" alt="Screen Shot 2022-01-18 at 2 32 55 PM" src="https://user-images.githubusercontent.com/43388/150029374-0d2897e2-a731-432a-aac8-d01f1380a1e9.png">

**Comment by heckj** [comment]
Initial value:
```
simd_float4x4(
    simd_float4(-1.1513867, -0.12829041, 2.2153668, 0.0),
    simd_float4(2.173691, 0.43699712, 1.155033, 0.0),
    simd_float4(-0.44651538, 2.458165, -0.08971556, 0.0),
    simd_float4(4.4409075, 13.833499, 8.220964, 1.0),
)
```

After transforms applied:
```
simd_float4x4(
    simd_float4(-1.0507754, -0.5739607, 2.194639, 0.0),
    simd_float4(2.173691, 0.43699712, 1.155033, 0.0),
    simd_float4(-0.64879775, 2.3936589, 0.315371, 0.0),
    simd_float4(4.4409075, 13.833499, 8.220964, 1.0),
)
```

**Comment by heckj** [comment]
fully resolved. example with sympodial trees
<img width="989" alt="Screen Shot 2022-01-22 at 1 53 34 PM" src="https://user-images.githubusercontent.com/43388/150656669-368740be-ba5e-48a0-be05-187877f10c6d.png">

and monopodial trees
<img width="927" alt="Screen Shot 2022-01-22 at 1 52 40 PM" src="https://user-images.githubusercontent.com/43388/150656671-000ccd10-918f-4a89-90ad-34009afdffc8.png">

**Comment by heckj** [comment]
fully resolved by #32
